+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+                    z_impulse = [z_impulse,{obj.w.Gamma,obj.w.Beta}];
+                    switch opts.conic_model_switch_handling
+                      case 'Plain'
+                        % no extra vars
+                      case 'Abs'
+                        z_alg = [z_alg,{obj.w.p_vt,obj.w.n_vt}];
+                        z_impulse = [z_impulse,{obj.w.P_vt,obj.w.N_vt}];
+                      case 'Lp'
+                        z_alg = [z_alg,{obj.w.p_vt,obj.w.n_vt,obj.w.alpha_vt}];
+                        z_impulse = [z_impulse,{obj.w.P_vt,obj.w.N_vt,obj.w.Alpha_vt}];
+                    end
+                end
+            end
+            if opts.lift_velocity_state
+                z_alg_f_x = [z_alg_f_x, {obj.w.z_v}]
+            end
+            obj.z_alg = vdx.VariableGroup(z_alg);
+            obj.z_impulse = vdx.VariableGroup(z_impulse);
+            obj.z_alg_f_x = vdx.VariableGroup(z_alg_f_x);
+        end
+
+        function generate_direct_transcription_constraints(obj)
+            import casadi.*
+            model = obj.model;
+            opts = obj.opts;
+            dcs = obj.dcs;
+            dims = obj.dcs.dims;
+
+            rbp = ~opts.right_boundary_point_explicit;
+
+            v_global = obj.w.v_global();
+            p_global = obj.p.p_global();
+
+            x_0 = obj.w.x(0,0,opts.n_s);
+            z_0 = obj.w.z(0,0,opts.n_s);
+            
+            obj.g.z(0,0,opts.n_s) = {dcs.g_z_fun(x_0, z_0, obj.w.u(1), v_global, [p_global;obj.p.p_time_var(1)])};
+            
+            x_prev = obj.w.x(0,0,opts.n_s);
+            for ii=1:opts.N_stages
+                h0 = obj.p.T().val/(opts.N_stages*opts.N_finite_elements(ii));
+                
+                ui = obj.w.u(ii);
+                p_stage = obj.p.p_time_var(ii);
+                p = [p_global;p_stage];
+                if obj.opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                    s_sot = obj.w.sot(ii);
+                elseif obj.opts.use_speed_of_time_variables
+                    s_sot = obj.w.sot();
+                else
+                    s_sot = 1;
+                end
+                if opts.time_optimal_problem && ~opts.use_speed_of_time_variables
+                    t_stage = obj.w.T_final()/(opts.N_stages*opts.N_finite_elements(ii));
+                elseif opts.time_optimal_problem
+                    t_stage = s_sot*obj.p.T()/opts.N_stages;
+                else
+                    t_stage = obj.p.T()/opts.N_stages;
+                end
+
+                sum_h = 0;
+                for jj=1:opts.N_finite_elements(ii)
+                    if obj.opts.use_fesd
+                        h = obj.w.h(ii,jj);
+                        sum_h = sum_h + h;
+                    elseif opts.time_optimal_problem && ~opts.use_speed_of_time_variables
+                        h = obj.w.T_final()/(opts.N_stages*opts.N_finite_elements(ii));
+                    else
+                        h = h0;
```

Comment:
if this is the first time where use_fesd is false, then invoke an error message and ask the user to set use_fesd = true, as discussed in a previous comment the false case does not make sense.